ecs-cli/modules/cli/servicediscovery/servicediscovery_helper.go

```diff
@@ -0,0 +1,290 @@
+
+// Imported Route53 Utility functions that can be mocked in tests
+// Adding the type signature allows code editor autocompletion and checking to work normally
+var findPrivateNamespace route53.FindPrivateNamespaceFunc = route53.FindPrivateNamespace
+var findPublicNamespace route53.FindPublicNamespaceFunc = route53.FindPublicNamespace
+
+// CreateFunc is the interface/signature for Create
+// This helps when writing code in other packages that need to mock Create (specifically it's a nicety that helps IDE features work)
+type CreateFunc func(networkMode, serviceName string, c *context.ECSContext) (*string, error)
+
+func resolveIntPointerFieldOverride(c *cli.Context, flagName string, ecsParamsVal *int64, field string) (*int64, error) {
+	flagVal, err := getInt64FromCLIContext(c, flagName)
+	if err != nil {
+		return nil, err
+	}
+
+	if flagVal != nil && ecsParamsVal != nil {
+		paramsVal := string(*ecsParamsVal)
+		override := string(*flagVal)
+		showFieldOverrideMsg(paramsVal, override, field)
+	}
+	if flagVal != nil {
+		return flagVal, nil
+	}
+	return ecsParamsVal, nil
+}
+
+// getInt64FromCLIContext reads the flag from the cli context and typecasts into *int64
+func getInt64FromCLIContext(c *cli.Context, flag string) (*int64, error) {
+	value := c.String(flag)
+	if value == "" {
+		return nil, nil
+	}
+	intValue, err := strconv.ParseInt(value, 10, 64)
+	if err != nil {
+		return nil, fmt.Errorf("Please pass integer value for the flag %s", flag)
+	}
+	return aws.Int64(intValue), nil
+}
+
+func resolveStringFieldOverride(c *cli.Context, flagName, ecsParamsVal string, field string) string {
+	flagVal := c.String(flagName)
+	if flagVal != "" && ecsParamsVal != "" {
+		showFieldOverrideMsg(ecsParamsVal, flagVal, field)
+	}
+	if flagVal != "" {
+		return flagVal
+	}
+	return ecsParamsVal
+}
+
+func showFieldOverrideMsg(val string, override string, field string) {
+	overrideMsg := "Using flag value as override (was %v but is now %v)"
+
+	logrus.WithFields(logrus.Fields{
+		"Service Discovery field": field,
+	}).Infof(overrideMsg, val, override)
+}
+
+// Validates that SD input does not container conflicting values
+// This function is used after the inputs from flags and ECS Params have been merged
+func validateMergedSDInputFields(input *utils.ServiceDiscovery, networkMode string) error {
+	if usesSRV(input, networkMode) && input.ContainerName == "" {
+		return fmt.Errorf("container_name is a required field when using SRV DNS records")
+	}
+	if usesSRV(input, networkMode) && input.ContainerPort == nil {
+		return fmt.Errorf("container_port is a required field when using SRV DNS records")
+	}
+
+	hasPublic := hasNamespace(input.PublicDNSNamespace.Namespace)
+	hasPrivate := hasNamespace(input.PrivateDNSNamespace.Namespace)
+
+	if hasPublic && hasPrivate {
+		return fmt.Errorf("both a public and private namespace can not be used with Service Discovery; please use only 1 namespace")
```

Comment:
Should we capitalize this error message?

Reply:
My editor says that go style suggests that error messages shouldn't be capitalized... but if we think it would look better for the user if this error is capitalized then I am cool with changing it. This is what the user would see:

```
$ ecs-cli compose service up --enable-service-discovery --private-dns-namespace blah --public-dns-namespace bleh
INFO[0000] Using ECS task definition                     TaskDefinition="dockerstats:1"
FATA[0000] both a public and private namespace can not be used with Service Discovery; please use only 1 namespace 
``

Reply:
Ok I capitalized it